src/audio/module_adapter/module/generic.c

```diff
@@ -225,9 +228,12 @@ int module_prepare(struct processing_module *mod)
 	return ret;
 }
 
-int module_process(struct processing_module *mod, struct input_stream_buffer *input_buffers,
-		   int num_input_buffers, struct output_stream_buffer *output_buffers,
-		   int num_output_buffers)
+/* for simplification, module process gets a void pointers as data types
+ * and calls proper handler at the end
+ */
+int module_process(struct processing_module *mod,
+		   void *sources, int num_of_sources,
+		   void *sinks, int num_of_sinks)
```

Comment:
`sources` and `sinks` should have their proper `struct source **` and `struct sink **` types with proper sparse annotations